### PR TITLE
Don't emit cannot move errors twice in migrate mode

### LINKED ISSUE #55154: nll suggestion about moving out of a `&T` is to make it a `&mut T`
```rust
struct Sheep {
    message: String
}

impl Sheep {
    fn talk(self) {
        println!("{}", self.message);
    }
}

fn main()
{
    let sheep = &Sheep { message: "Määh".into() };
    sheep.talk();
}
```

([Playground](https://play.rust-lang.org/?version=beta&mode=debug&edition=2018&gist=e7954e7e7039e4ecb5996909069beb12))

Errors:

```
   Compiling playground v0.0.1 (/playground)
error[E0507]: cannot move out of borrowed content
  --> src/main.rs:14:5
   |
14 |     sheep.talk();
   |     ^^^^^ cannot move out of borrowed content

error[E0507]: cannot move out of `*sheep` which is behind a `&` reference
  --> src/main.rs:14:5
   |
13 |     let sheep = &Sheep { message: "Määh".into() };
   |                 --------------------------------- help: consider changing this to be a mutable reference: `&mut Sheep { message: "Määh".into() }`
14 |     sheep.talk();
   |     ^^^^^
   |     |
   |     cannot move out of `*sheep` which is behind a `&` reference
   |     `sheep` is a `&` reference, so the data it refers to cannot be moved

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0507`.
error: Could not compile `playground`.

To learn more, run the command again with --verbose.

```

### PR DESCRIPTION
**Pull request by matthewjasper**
Closes #55154 
cc #53004

r? @pnkfelix

### CODE REVIEW

File: src/librustc_mir/borrow_check/mod.rs [hunk 1]
```diff
@@ -1831,7 +1831,10 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
             | Write(wk @ WriteKind::StorageDeadOrDrop)
             | Write(wk @ WriteKind::MutableBorrow(BorrowKind::Shared))
             | Write(wk @ WriteKind::MutableBorrow(BorrowKind::Shallow)) => {
-                if let Err(_place_err) = self.is_mutable(place, is_local_mutation_allowed) {
+                if let (Err(_place_err), true) = (
+                    self.is_mutable(place, is_local_mutation_allowed),
+                    self.errors_buffer.is_empty()
```
Comment (pnkfelix) [inline src/librustc_mir/borrow_check/mod.rs]: Ah great idea to let the ICE through if we have already signaled errors. Or at least I hope it’s a great idea.

**Review by pnkfelix — APPROVED** [review]
@bors r+

**Comment by pnkfelix** [comment]
Ugh sorry I had used my phone and I guess the review comment system there doesn't bubble out to here in a manner that bors can see!

**Comment by pnkfelix** [comment]
Ugh sorry I had used my phone and I guess the review comment system there doesn't bubble out to here in a manner that bors can see!

**Comment by matthewjasper** [comment]
CI doesn't appear to be actually running. does
@bors retry
reset it?

**Comment by Aaron1011** [comment]
It looks like Bors might be stuck - all commits on the `auto` branch have their CI checks passing.

**Comment by kennytm** [comment]
@bors retry

3 hour timeout

**Pull request merged**